component-catalog/index.html

```diff
@@ -18,56 +18,149 @@
 
       // HIGHLIGHTER
 
-      // start listening to events when the user starts dragging/clickingdown
-      app.ports.highlighterListen.subscribe(function (id) {
+      app.ports.highlighterInit.subscribe(([id, scrollFriendly, readonly]) => {
+        console.log("highlighterInit", id, scrollFriendly, readonly);
+
+        // TODO: do we need to prevent multiple initializations?
+
+        const touchMoveListener = handleTouchEvents("move", id);
         window.requestAnimationFrame(() => {
-          var highlighter = document.getElementById(id);
+          const highlighter = document.getElementById(id);
           if (!id || !highlighter) {
-            // in a real application, report an error at this point.
+            console.log("highlighter not found", id);
             return;
           }
 
-          document.addEventListener("mouseup", onDocumentUp(id));
-          document.addEventListener("touchend", onDocumentUp(id));
+          highlighter.addEventListener("mousedown", (_) => {
+            document.addEventListener("mouseup", onDocumentUp(id, "mouse"), {
+              once: true,
+            });
+          });
+          highlighter.addEventListener("touchstart", (_) => {
+            console.log("touchstart", id);
```

Comment:
There's a few console.logs here. Are they all being kept deliberately or left over by mistake?

Reply:
Deliberately, they're also kept in the real monolith, to help us handle issue reports, so we can ask people "can you repro and show the console output?"